daemon/emer-boot-id-provider.c

```diff
@@ -280,3 +280,48 @@ emer_boot_id_provider_get_id (EmerBootIdProvider *self,
   id_is_valid = TRUE;
   return TRUE;
 }
+
+#define KERNEL_CMDLINE_PATH	"/proc/cmdline"
+#define LIVE_BOOT_FLAG_REGEX	"\\bendless\\.live_boot\\b"
+#define DUAL_BOOT_FLAG_REGEX	"\\bendless\\.image\\.device\\b"
+
+#define NORMAL_BOOT	0x0
+#define DUAL_BOOT	0x1
+#define LIVE_BOOT	0x2
+
+/*
+ * emer_boot_id_provider_get_boot_type:
+ *
+ * Get boot type by checking kernel's boot command line.
+ *
+ * Returns: a number 0x0 for normal boot, 0x1 for dual boot, 0x2 for live boot.
+ */
+guint8
+emer_boot_id_provider_get_boot_type (void)
+{
+  g_autofree gchar *cmdline = NULL;
+  g_autoptr(GError) error = NULL;
+  static guchar boot_type = NORMAL_BOOT;
```

Comment:
Can you also make this `guint8` to match the return type?

Reply:
ok